lib/rules/no-will-update-set-state.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * @fileoverview Prevent usage of setState in componentWillUpdate
+ * @author Yannick Croissant
+ */
+'use strict';
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prevent usage of setState in componentWillUpdate',
+      category: 'Best Practices',
+      recommended: false
```

Comment:
Adding things to the recommended rules is a breaking change, so it should be in a separate PR regardless.